lib/msal-browser/src/cache/TokenCache.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccessTokenEntity, ICrypto, IdTokenEntity, Logger, ScopeSet, Authority, AuthorityOptions, ServerAuthorizationROPCResponse } from "@azure/msal-common";
+import { BrowserConfiguration, buildConfiguration, Configuration } from "../config/Configuration";
+import { SilentRequest } from "../request/SilentRequest";
+import { BrowserCacheManager } from "./BrowserCacheManager";
+import { ITokenCache } from "./ITokenCache";
+import { BrowserAuthError } from "../error/BrowserAuthError";
+
+export type LoadTokenOptions = {
+    clientInfo?: string,
+    extendedExpiresOn?: number,
+    callback?: TokenCallback
+};
+
+export type TokenCallback = (key: string, value: string) => void;
+
+/**
+ * Token cache manager
+ */
+export class TokenCache implements ITokenCache {
+    // Flag to indicate if in browser environment
+    public isBrowserEnvironment: boolean;
+    // Input configuration by developer/user
+    protected config: BrowserConfiguration;
+    // Browser cache storage
+    private storage: BrowserCacheManager;
+    // Logger
+    private logger: Logger;
+    // Crypto class
+    private cryptoObj: ICrypto;
+
+    constructor(configuration: Configuration, storage: BrowserCacheManager, logger: Logger, cryptoObj: ICrypto) {
+        this.isBrowserEnvironment = typeof window !== "undefined";
+        this.config = buildConfiguration(configuration, this.isBrowserEnvironment);
+        this.storage = storage;
+        this.logger = logger;
+        this.cryptoObj = cryptoObj;
+    }
+    
+    // Move getAllAccounts here and cache utility APIs
+
+    /**
+     * API to load tokens to msal-browser cache. 
+     * @param request 
+     * @param response
+     * @param options
+     */
+    loadTokens(request: SilentRequest, response: ServerAuthorizationROPCResponse, options: LoadTokenOptions): void {
+        this.logger.info("TokenCache - loadTokens called");
+
+        if (!response.id_token) {
+            throw BrowserAuthError.createUnableToLoadTokenError("Please ensure server response includes id token.");
+        }
+
+        if (request.account) {
+            this.loadIdToken(response.id_token, request.account.homeAccountId, request.account.environment, request.account.tenantId, options);
+            this.loadAccessToken(request, response, options, request.account.homeAccountId, request.account.environment, request.account.tenantId);
```

Comment:
nit: Can we manage the same order for common parameters? Increases readability.

Reply:
Done.